src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
```

Comment:
Wow, it seems to be even worse:
```javascript
console.time('string');
var str = '';
for (var i = 0; i < 1000000; i++) {
  str += 'The line with ' + i + '\n';
}
var all = str;
console.timeEnd('string');
```
`node performance1.js` done 10x: 173ms+/-2ms

```javascript
console.time('buffer');
var str = [];
for (var i = 0; i < 1000000; i++) {
  str.push('The line with ' + i);
}
var all = str.join('\n');
console.timeEnd('buffer');
```
`node performance2.js` done 10x: 316ms+/-10ms

Add the overhead of `new` and method calls...

Is that what those little examples show what you mean here?